kfs-core/src/main/java/edu/arizona/kfs/vnd/document/service/impl/VendorServiceImpl.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+    public boolean isVendorForeign(Integer vendorHeaderGeneratedIdentifier) {
+        return vendorService.isVendorForeign(vendorHeaderGeneratedIdentifier);
+    }
+
+    public boolean isVendorInstitutionEmployee(Integer vendorHeaderGeneratedIdentifier) {
+        return vendorService.isVendorInstitutionEmployee(vendorHeaderGeneratedIdentifier);
+    }
+
+    public boolean noRouteSignificantChangeOccurred(VendorDetail newVDtl, VendorHeader newVHdr, VendorDetail oldVDtl, VendorHeader oldVHdr) {
+        return vendorService.noRouteSignificantChangeOccurred(newVDtl, newVHdr, oldVDtl, oldVHdr);
+    }
+
+    public void saveVendorHeader(VendorDetail vendorDetail) {
+        vendorService.saveVendorHeader(vendorDetail);
+    }
+    
+	public List<Note> getVendorNotes(VendorDetail vendorDetail) {
+		return vendorService.getVendorNotes(vendorDetail);
+	}
+
+	public boolean isVendorContractExpired(Document document, Integer vendorContractGeneratedIdentifier, VendorDetail vendorDetail) {
+		return vendorService.isVendorContractExpired(document, vendorContractGeneratedIdentifier, vendorDetail);
+	}
+	
+	public VendorAddress getVendorDefaultAddress(Integer vendorHeaderId, Integer vendorDetailId, String addressType, String campus, boolean activeCheck) {
+		return vendorService.getVendorDefaultAddress(vendorHeaderId, vendorDetailId, addressType, campus, activeCheck);
+	}   
+
+    /**
+     * @see org.kuali.kfs.vnd.document.service.VendorService#shouldVendorRouteForApproval(java.lang.String)
+     * Returns true if super class returns true, or if there was a change in an extended attribute.
+     */
+    public boolean shouldVendorRouteForApproval(String documentId) {        
+        if (getVendorService().shouldVendorRouteForApproval(documentId)) {
+             return true;
+        }
+                
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Entering customized shouldVendorRouteForApproval.");
+        }
+        
+        boolean shouldRoute = true;
+        MaintenanceDocument newDoc = null;
+        try {
+            newDoc = (MaintenanceDocument) getDocumentService().getByDocumentHeaderId(documentId);
+        }
+        catch (WorkflowException we) {
+            throw new RuntimeException("A WorkflowException was thrown which prevented the loading of " + "the comparison document (" + documentId + ")", we);
+        }
+
+        if (ObjectUtils.isNotNull(newDoc)) {
+
+            VendorDetailExtension oldVDtlExtension =  (VendorDetailExtension) ((VendorDetail) newDoc.getOldMaintainableObject().getBusinessObject()).getExtension();
+            VendorDetailExtension newVDtlExtension = (VendorDetailExtension) ((VendorDetail)newDoc.getNewMaintainableObject().getBusinessObject()).getExtension();
+
+            if (ObjectUtils.isNotNull(oldVDtlExtension)) {
+                shouldRoute = !(noRouteSignificantChangeOccurredForExtensions(oldVDtlExtension, newVDtlExtension));
```

Comment:
Call to method noRouteSignificantChangeOccurredForExtensions has oldVDtlExtension is first in parameter list. In noRouteSignificantChangeOccurredForExtensions method declaration, newVDtlExtension is first in parameter list.